File: source/algorithm/NQ021/NQ021_improved.cpp
//NQ021 废强重建
#include <iostream>
using namespace std;
const int N=107;
int main(){
    int n,a[N];
    while(cin>>n,n>0){
        int avg=0;
        for(int i=0;i<n;i++){
            cin>>a[i];
            avg+=a[i];
        }
        avg/=n;  // 计算平均值
        int res=0;
        for(int i=0;i<n;i++){
            if(a[i]<avg)res+=avg-a[i];  // 统计空缺数
        }
        cout<<res<<'\n'<<'\n';  // 输出结果，空行分隔
    }
    return 0;
}

File: source/algorithm/NQ021/NQ021_improved.py
# NQ021 废强重建
import sys
first=True
while True:
    line=sys.stdin.readline()
    if not line:break
    n=int(line.strip())
    if n==0:break
    h=[]
    while len(h)<n:
        hl=sys.stdin.readline()
        if not hl:break
        h.extend(map(int,hl.strip().split()))
    total=sum(h)
    avg=total//n  # 计算平均值（向下取整）
    need=sum(avg-ht for ht in h if ht<avg)  # 统计需要的砖块数
    if not first:print()
    print(need)
    first=False